lib/uktt/base.rb

```diff
@@ -10,6 +11,12 @@ def retrieve
     raise NotImplementedError
   end
 
+  def config=(new_opts = {})
+    merged_opts = Uktt.config.merge(new_opts)
+    Uktt.configure(merged_opts)
+    @config = Uktt.config
```

Comment:
Hmm this looks interesting. So what is the difference between `Uktt.configure(merged_opts)` and `Uktt.config` ? I assume `Uktt.configure(merged_opts)` is doing some sort of mutation, but @config receives Uktt.config. A bit confusing imo.

Reply:
From what I understand:

```ruby
  @config = Uktt.config
```
Is creating a variable in the resource wrapper instance that references another variable that lives as a global in the Uktt singleton.

This is actually a good place to call `.deep_dup` to fix our issue of updating a global bit of config when making multiple requests and experiencing that race condition we talked about.

Reply:
I agree it's confusing, tbh

Reply:
This is a big thing I want to remove/replace

Reply:
Madness! Ok, thanks for the context. I guess we can add it to our list of refactors.

Reply:
I think I'd probably move to configuration that's global for this client and configuration that's specific to the request. This gem mixes those two things and tries to prevent mixing the two by having a list of valid values we can configure behind the valid config keys 

``` ruby
  @valid_config_keys = @config.keys

  # Configure through hash
  def self.configure(opts = {})
    opts.each { |k, v| @config[k.to_sym] = v if @valid_config_keys.include? k.to_sym }
  end
```

I think I'd like to have a client/connection concept (similar to faraday) and then request-specific configuration is totally separate/passed into the methods that make the request.